metrics.go

```diff
@@ -0,0 +1,32 @@
+package caddy
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// define and register the metrics used in this package.
+func init() {
+	initAdminMetrics()
+	prometheus.MustRegister(prometheus.NewBuildInfoCollector())
+
+	// TODO: could be useful (import "github.com/povilasv/prommod")
+	// prometheus.MustRegister(prommod.NewCollector(ns))
```

Comment:
Let's try to decide if this is useful. Can you help me understand what this does?

Reply:
It adds a bunch of extra informational metrics labeled by the list of modules included in the binary, and versions. It's basically just a way to track which versions of which modules are in use. We use this at work with some of our Go services to help us track which modules are being used. Occasionally useful if there's a vulnerability found in one of them to see which services are affected.

Here's some example output from their README:

```
# HELP go_mod_info A metric with a constant '1' value labeled by dependency name, version, from which app_name was built.
# TYPE go_mod_info gauge
go_mod_info{name="github.com/beorn7/perks",program="app_name",version="v0.0.0-20180321164747-3a771d992973"} 1
go_mod_info{name="github.com/golang/protobuf",program="app_name",version="v1.2.0"} 1
go_mod_info{name="github.com/matttproud/golang_protobuf_extensions",program="app_name",version="v1.0.1"} 1
go_mod_info{name="github.com/povilasv/prommod",program="app_name",version="v0.0.11-0.20190309143328-e661980fc053"} 1
go_mod_info{name="github.com/prometheus/client_golang",program="app_name",version="v0.9.2"} 1
go_mod_info{name="github.com/prometheus/client_model",program="app_name",version="v0.0.0-20180712105110-5c3871d89910"} 1
go_mod_info{name="github.com/prometheus/common",program="app_name",version="v0.0.0-20181126121408-4724e9255275"} 1
go_mod_info{name="github.com/prometheus/procfs",program="app_name",version="v0.0.0-20181204211112-1dc9a6cbc91a"} 1
```

It's a super-cheap metric to add since it gets set once at startup and never changes.

Reply:
Nope! Sounds like you're good with it, so I'll add this 😉

Reply:
I'm not sure this is all that useful for real-time monitoring. I find it interesting, but the normal Prometheus build info metric (`https://github.com/prometheus/common/version`) is usually more than enough. Being able to tie an instance back to a build is one thing, but this seems like too much granularity.

For Caddy, this is 26+ additional metrics.

Reply:
Yeah, the extra metrics is mostly why I wasn't sure about adding this.

I'll leave this out for now, and we can revisit later.

Reply:
```suggestion
```